src/components/data/listing/Listing.js

```diff
@@ -37,53 +38,23 @@ function Listing(props) {
     }, [path]);
 
     useEffect(() => {
-        setLoading(true);
-        fetch(
-            `/api/filesystem/paged-directory?path=${path}&limit=${rowsPerPage}&sort-col=${orderBy}&sort-dir=${order}&offset=${rowsPerPage *
+        callApi({
+            endpoint: `/api/filesystem/paged-directory?path=${path}&limit=${rowsPerPage}&sort-col=${orderBy}&sort-dir=${order}&offset=${rowsPerPage *
                 page}`,
-            {
-                method: "GET",
-                credentials: "include",
-            }
-        )
-            // Do something with errors from terrain.
-            .then((resp) => {
-                if (resp.status < 200 || resp.status > 299) {
-                    setLoading(false);
-                    throw Error(resp);
-                }
-                return resp;
-            })
-
-            .then((resp) => resp.json())
-
-            // Unify the data listing, adding type info along the way.
-            .then(
-                (respData) => {
-                    return {
-                        total: respData.total,
-                        listing: [
-                            ...respData.folders.map((f) => ({
-                                ...f,
-                                type: "FOLDER",
-                            })),
-                            ...respData.files.map((f) => ({...f, type: "FILE"})),
-                        ].map((i) => camelcaseit(i)) // camelcase the fields for each object, for consistency.
-                    }
-                })
-
-            // Storing the listing here avoids having to regen the
-            // list of items on every render pass.
-            .then((resp) => {
-                setData(resp);
-                setLoading(false);
-            })
-
-            // We need to figure out a consistent error handler.
-            .catch((e) => {
-                console.log(`error ${e.message}`)
-                setLoading(false);
-            });
+            setLoading,
+        }).then((respData) => {
+            respData &&
+                setData({
+                    total: respData?.total,
```

Comment:
Since there's already a `respData && setData` check above, does that make these `respData?.` checks redundant?